src/libtools/reducedgraph.cc

```diff
@@ -0,0 +1,182 @@
+/*
+ *            Copyright 2009-2018 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <cassert>
+#include <votca/tools/edge.h>
+#include <votca/tools/reducedgraph.h>
+
+using namespace std;
+
+namespace votca {
+namespace tools {
+
+class GraphNode;
+
+bool compareChainWithChains_(vector<int> chain, vector<vector<int>> chains){
```

Comment:
why copies?

Reply:
Because I'm not a master programmer yet, :)